### PR TITLE
[mpeg2d] Fix for a stream with invalid temporal_reference count

### PR DESCRIPTION
**Pull request by IlyaSergeyev**
Correct submitting of first I-frame top part

testcase: tractor_422_1920x1080i_50f.mpg

### CODE REVIEW

File: _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp [hunk 1]
```diff
@@ -874,12 +883,9 @@ namespace UMC_MPEG2_DECODER
             return true; // Full frame
         }
 
-        // field picture
-        const auto firstFrameSlice = m_currFrame->GetAU(0)->GetSlice(0);
-        const auto newPicHdr = *m_currHeaders.picHdr.get();
         const auto newPicExtHdr = *m_currHeaders.picExtHdr.get();
 
-        if (IsFieldOfOneFrame(firstFrameSlice->GetPicHeader(), firstFrameSlice->GetPicExtHeader(), newPicHdr, newPicExtHdr))
+        if (IsFieldOfSameFrame())
```
Comment (dmitryermilov) [inline _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp]: I know I previously wrote that member function is ok. Thinking more, now I can't say so. "Is field of one frame" - what field and what frame? Previously it was clear from arg list

File: _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp [hunk 2]
```diff
@@ -848,12 +848,21 @@ namespace UMC_MPEG2_DECODER
...
     {
-        if (picHdr.temporal_reference != newPictureHdr.temporal_reference)  // 6.3.9
-            return false;
+        const auto firstFrameSlice = m_currFrame->GetAU(0)->GetSlice(0);
+        const auto picHdr = firstFrameSlice->GetPicHeader();
+        const auto picExtHdr = firstFrameSlice->GetPicExtHeader();
+        const auto newPicHdr = *m_currHeaders.picHdr.get();
+        const auto newPicExtHdr = *m_currHeaders.picExtHdr.get();
+
+        // for I-frame we don't need to check if temporal refs are eq
```
Comment (dmitryermilov) [inline _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp]: need to highlighting that this is WA.  "for I-frame we don't need to check" - this is wrong, we need to check!

File: _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp (hunk 1 above)
Comment (IlyaSergeyev) [reply _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp]: Maybe we need to use a different name?

File: _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp (hunk 2 above)
Comment (IlyaSergeyev) [reply _studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp]: ok

**Pull request merged**